### ISSUE TITLE
Add Windows CI

### ISSUE THREAD

**Issue by alexeagle** [bazelbuild/rules_typescript#56]
And make it green!

**Comment by filipesilva** [comment]
Running into some issues with `gazello`: https://github.com/bazelbuild/rules_go/issues/736#issuecomment-346832619

**Comment by alexeagle** [comment]
Is `gazelle` the only problem? I think I'm going to pull that out, because the go code is shared with google internal so we'll use `glaze` instead.

**Comment by filipesilva** [comment]
I can't say if it's the only one, it is blocking my progress so I couldn't move further.

**Comment by alexeagle** [comment]
Can you just not run `gazelle`? It's only there to update the BUILD.bazel
files, which you probably don't need to do unless you're making Go changes

**Comment by filipesilva** [comment]
How do I do that?

What I'm doing currently is:
- `bazel run @yarn//:yarn`
- `bazel build ...`

```
kamik@T460p MINGW64 D:/sandbox/rules_typescript_2 (master)
$ bazel run @yarn//:yarn
.................
INFO: Analysed target @yarn//:yarn (7 packages loaded).
INFO: Found 1 target...
Target @yarn//:yarn up-to-date:
  C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/external/yarn/yarn
  C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/external/yarn/yarn.exe
INFO: Elapsed time: 49.614s, Critical Path: 0.04s
INFO: Build completed successfully, 5 total actions

INFO: Running command line: C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/external/yarn/yarn.exe
yarn install v0.22.0
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 9.14s.

kamik@T460p MINGW64 D:/sandbox/rules_typescript_2 (master)
$ bazel build ...
DEBUG: C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/external/io_bazel_rules_go/go/private/toolchain.bzl:104:5: mkdir C:\WINDOWS\go-build894015291: Access is denied.
ERROR: C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/external/io_bazel_rules_go/go/toolchain/BUILD.bazel:6:1: every rule of type _go_toolchain implicitly depends upon the target '@go_sdk//:packages.txt', but this target could not be found because of: no such package '@go_sdk//': Traceback (most recent call last):
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/external/io_bazel_rules_go/go/private/toolchain.bzl", line 45
                _prepare(ctx)
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/external/io_bazel_rules_go/go/private/toolchain.bzl", line 105, in _prepare
                fail("failed to list standard package...")
failed to list standard packages
ERROR: Analysis of target '//internal/devserver/main:main' failed; build aborted: no such package '@go_sdk//': Traceback (most recent call last):
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/external/io_bazel_rules_go/go/private/toolchain.bzl", line 45
                _prepare(ctx)
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/external/io_bazel_rules_go/go/private/toolchain.bzl", line 105, in _prepare
                fail("failed to list standard package...")
failed to list standard packages
INFO: Elapsed time: 119.614s
FAILED: Build did NOT complete successfully (43 packages loaded)

kamik@T460p MINGW64 D:/sandbox/rules_typescript_2 (master)
$ bazel version
Build label: 0.8.0
Build target: bazel-out/msvc_x64-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Mon Nov 27 13:18:30 2017 (1511788710)
Build timestamp: 1511788710
Build timestamp as int: 1511788710
```

**Comment by filipesilva** [comment]
Following a conversation with @alexeagle, I followed these steps:
- commented out `go_rules` related statements from `WORKSPACE`
- commented out `gazelle` related statements from `BUILD.bazel`
- ran `bazel clean --expunge`
- ran `bazel build internal/tsc_wrapped`

The last command returned an error that seems related to windows environments (it is a paths problem):
```
kamik@T460p MINGW64 /d/sandbox/rules_typescript_2 (master)
$ bazel build internal/tsc_wrapped
..................
INFO: Analysed target //internal/tsc_wrapped:tsc_wrapped (12 packages loaded).
INFO: Found 1 target...
ERROR: D:/sandbox/rules_typescript_2/internal/tsetse/rules/BUILD:19:1: Compiling TypeScript (devmode) //internal/tsetse/rules:rules failed (Exit 1)
error TS5033: Could not write file 'C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/internal/tsetse/checker.js': EPERM: operation not permitted, open 'C:\users\kamik\appdata\local\temp\_bazel_kamik\5xzmk4di\execroot\build_bazel_rules_typescript\bazel-out\x64_windows-fastbuild\bin\internal\tsetse\checker.js'.
error TS5033: Could not write file 'C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/internal/tsetse/error_code.js': EPERM: operation not permitted, open 'C:\users\kamik\appdata\local\temp\_bazel_kamik\5xzmk4di\execroot\build_bazel_rules_typescript\bazel-out\x64_windows-fastbuild\bin\internal\tsetse\error_code.js'.
error TS5033: Could not write file 'C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/internal/tsetse/failure.js': EPERM: operation not permitted, open 'C:\users\kamik\appdata\local\temp\_bazel_kamik\5xzmk4di\execroot\build_bazel_rules_typescript\bazel-out\x64_windows-fastbuild\bin\internal\tsetse\failure.js'.
error TS5033: Could not write file 'C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/internal/tsetse/rule.js': EPERM: operation not permitted, open 'C:\users\kamik\appdata\local\temp\_bazel_kamik\5xzmk4di\execroot\build_bazel_rules_typescript\bazel-out\x64_windows-fastbuild\bin\internal\tsetse\rule.js'.
error TS5055: Cannot write file 'C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/internal/tsetse/checker.d.ts' because it would overwrite input file.
error TS5055: Cannot write file 'C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/internal/tsetse/error_code.d.ts' because it would overwrite input file.
error TS5055: Cannot write file 'C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/internal/tsetse/failure.d.ts' because it would overwrite input file.
error TS5055: Cannot write file 'C:/users/kamik/appdata/local/temp/_bazel_kamik/5xzmk4di/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/internal/tsetse/rule.d.ts' because it would overwrite input file.
Target //internal/tsc_wrapped:tsc_wrapped failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 51.411s, Critical Path: 10.
```

**Comment by filipesilva** [comment]
Just tried again with everything updated and there's a different set of errors:
```
kamik@T460p MINGW64 /d/sandbox/rules_typescript (master)
$ bazel clean --expunge
.
INFO: Starting clean (this may take a while). Consider using --expunge_async if the clean takes more than several minutes.

kamik@T460p MINGW64 /d/sandbox/rules_typescript (master)
$ bazel run @yarn//:yarn
..................
INFO: Analysed target @yarn//:yarn (7 packages loaded).
INFO: Found 1 target...
Target @yarn//:yarn up-to-date:
  C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/external/yarn/yarn
  C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/external/yarn/yarn.exe
INFO: Elapsed time: 50.125s, Critical Path: 0.72s
INFO: Build completed successfully, 5 total actions

INFO: Running command line: C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/execroot/build_bazel_rules_typescript/bazel-out/x64_windows-fastbuild/bin/external/yarn/yarn.exe
yarn install v1.3.2
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.28s.

kamik@T460p MINGW64 /d/sandbox/rules_typescript (master)
$ bazel build internal/tsc_wrapped
ERROR: D:/sandbox/rules_typescript/internal/tsc_wrapped/BUILD.bazel:27:1: no such package '@build_bazel_rules_typescript_deps//': Traceback (most recent call last):
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/external/build_bazel_rules_nodejs/internal/npm_install.bzl", line 36
                repository_ctx.execute([repository_ctx.path(repository_...("")])
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/external/build_bazel_rules_nodejs/internal/npm_install.bzl", line 37, in repository_ctx.execute
                repository_ctx.path(repository_ctx.attr._npm)
Not a regular file: C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/external/nodejs/bin/npm and referenced by '//internal/tsc_wrapped:tsc_wrapped'
ERROR: D:/sandbox/rules_typescript/internal/tsc_wrapped/BUILD.bazel:27:1: no such package '@build_bazel_rules_typescript_deps//': Traceback (most recent call last):
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/external/build_bazel_rules_nodejs/internal/npm_install.bzl", line 36
                repository_ctx.execute([repository_ctx.path(repository_...("")])
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/external/build_bazel_rules_nodejs/internal/npm_install.bzl", line 37, in repository_ctx.execute
                repository_ctx.path(repository_ctx.attr._npm)
Not a regular file: C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/external/nodejs/bin/npm and referenced by '//internal/tsc_wrapped:tsc_wrapped'
ERROR: Analysis of target '//internal/tsc_wrapped:tsc_wrapped' failed; build aborted: no such package '@build_bazel_rules_typescript_deps//': Traceback (most recent call last):
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/external/build_bazel_rules_nodejs/internal/npm_install.bzl", line 36
                repository_ctx.execute([repository_ctx.path(repository_...("")])
        File "C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/external/build_bazel_rules_nodejs/internal/npm_install.bzl", line 37, in repository_ctx.execute
                repository_ctx.path(repository_ctx.attr._npm)
Not a regular file: C:/users/kamik/appdata/local/temp/_bazel_kamik/y9xg3tkg/external/nodejs/bin/npm
INFO: Elapsed time: 0.568s
FAILED: Build did NOT complete successfully (5 packages loaded)

kamik@T460p MINGW64 /d/sandbox/rules_typescript (master)
$ bazel version
Build label: 0.8.1
Build target: bazel-out/x64_windows-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Dec 5 14:29:39 2017 (1512484179)
Build timestamp: 1512484179
Build timestamp as int: 1512484179
```

**Comment by gregmagolan** [comment]
Fixes for Windows CI are under #155. `bazel build ...` and `bazel test ...` now working under Windows.

**Comment by alexeagle** [comment]
I think we have this now
